### ISSUE TITLE
weight use for loss function

### ISSUE THREAD

**Issue by xxxfrank** [PkuRainBow/OCNet#14]
![image](https://user-images.githubusercontent.com/34808562/46053030-aea13380-c173-11e8-9b95-ea2d8eb2fa2e.png)
i wonder how to calculate the weight  if i train with another dataset.
(i have tried 1/pixel frequency of each class ,but it works bad : (

Thanks

**Comment by PkuRainBow** [comment]
@xxxfrank Hi, we compute the weights by 1/log(pixel_count).

**Comment by eugenelawrence** [comment]
@PkuRainBow What do the pixel_count mean? Is it the pixel count of the specific class across the dataset? If so, the pixel_count corresponds to loss_weight 0.8373 should be exp(1/0.8373)=3.3013. It's weird!
Correct me if I have anywhere misunderstood.

**Comment by PkuRainBow** [comment]
@eugenelawrence 

Thanks for your interest in our work.

We will update the explaination of these weights in our update paper.

Please keep waiting for the arXiv update. We will notice you in the earliest time.

**Comment by PkuRainBow** [comment]
> @PkuRainBow
> I am wondering would you release the code for calculating the loss weights?

Of course if you need.

**Comment by KeyKy** [comment]
@PkuRainBow So, How did you compute the loss weight?

**Comment by PkuRainBow** [comment]
@KeyKy Here we employ the log function firstly,

![image](https://user-images.githubusercontent.com/4639578/50638519-003e3300-0f99-11e9-8b51-de336d8458f8.png)

Then, we will apply L1-normalize over all the log weights to act as the final weights. 

Here I provide the statistics of the pixels' count on the Cityscapes' training set.

![image](https://user-images.githubusercontent.com/4639578/50638579-40051a80-0f99-11e9-8a70-c39083d78c7e.png)

**Comment by yyfyan** [comment]
Hi, @PkuRainBow 
I used the code to Cal. but this is defferent.
`D = [2.01e+9,2.98e+8,9.96e+8,3.39e+7,4.50e+7,6.54e+7,
		9.57e+7,2.62e+7,7.21e+8,5.92e+7,1.45e+8,8.21e+7,
			1.00e+7,4.13e+8,1.45e+7,1.28e+7,1.45e+7,5.64e+6,
			2.57e+7]
			
w = np.log(D)/np.log(19)
print(w)
w_sum = np.sum(w)
print(w_sum)
w = w/w_sum
print(w)`

**Comment by PkuRainBow** [comment]
![image](https://user-images.githubusercontent.com/4639578/64302764-b51a5c80-cf39-11e9-98e7-2ea1281e0ed4.png)

**Comment by Anikily** [comment]
@PkuRainBow Could you tell me how to calculate the pixel_count on training dataset?

**Comment by AndyChang666** [comment]
@PkuRainBow Could you release the code of calculating class_freq in the whole dataset? Thank you.

**Comment by jianlong-yuan** [comment]
any news?
